src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -38,6 +44,12 @@ private sealed class EasyRequest : TaskCompletionSource<HttpResponseMessage>
             internal SendTransferState _sendTransferState;
             internal bool _isRedirect = false;
 
+            private SafeCallbackHandle _headerCallbackHandle;
+            private SafeCallbackHandle _sendCallbackHandle;
+            private SafeCallbackHandle _receiveBodyCallback;
+            private SafeCallbackHandle _seekCallbackHandle;
+            private SafeCallbackHandle _sslCtxCallbackHandle;
```

Comment:
We'd need a finalizer somewhere that would handling that free'ing if it wasn't explicitly Dispose'd, right?  I do think that'd be preferable to the current code, as even if we allocated a finalizable object (maybe even a SafeHandle) to store the five pointers, one is better than 5.